Assignment_2-data-emma-luk.js

```diff
@@ -0,0 +1,101 @@
+﻿// Emma Luk
+// Assignments 2 Using the revealing module pattern, build a Date utility module 
+
+var myDate = (function iife() {
+    var displaysModuleDate = null;
+
+    //private function: date instantiation
+    function _initDate() {
+        return new Date();
+    }
+
+    //check here that date is set: if not, initialize date to now
+    function setDate(date) {
+        if (date === undefined && null) {
+            displaysModuleDate = _initDate();
+        }
+        else if (typeof date === 'number' && !isNaN(date)) {

```

Comment:
This will work in most cases, but you should omit the second part of this check as the first part `typeof date === 'number` is sufficient. Checking by using `!isNaN(date)` can have unexpected effects. For example, if you were to pass the values true or false to setDate, it wouldn't behave like you may expect.